instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java

```diff
@@ -35,22 +36,27 @@
   private static final Cache<String, AttributeKey<String>> mapMessageAttributeKeyCache =
       Cache.bounded(100);
 
+  private static final AttributeKey<String> LOG_MARKER = AttributeKey.stringKey("log4j.marker");
```

Comment:
Logback has markers too -- do we want to encode library name in the attribute name? I'm worried that we'll run into a situation similar to #5765, perhaps it'd be better to avoid it right from the start and just name it `log.marker` or sth similar.

Reply:
@mateuszrzeszutek I gave contradictory advice here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6652#discussion_r974864036

> so that it doesn't look like a spec'd attribute

I'm not sure what the best compromise here is

Reply:
It looks like we're already prefixing MDC/context data with the library name (`log4j.context_data...`), so I suppose doing the same for markers is not out of place.
I guess we can merge it as it is -- we'll most likely have to rename things anyway once the log semantic conventions start to take shape.